src/main/java/org/opensearch/security/authtoken/jwt/ExpiringBearerAuthToken.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+package org.opensearch.security.authtoken.jwt;
+
+import java.util.Date;
+
+import org.opensearch.identity.tokens.BearerAuthToken;
+
+public class ExpiringBearerAuthToken extends BearerAuthToken {
+    private final String subject;
+    private final Date expiry;
+    private final long expiresInSeconds;
+
+    public ExpiringBearerAuthToken(final String serializedToken, final String subject, final Date expiry, final long expiresInSeconds) {
+        super(serializedToken);
+        this.subject = subject;
+        this.expiry = expiry;
+        this.expiresInSeconds = expiresInSeconds;
+    }
+
+    public String getSubject() {
+        return subject;
+    }
+
+    public Date getExpiry() {
+        return expiry;
+    }
+
+    public long getExpiresInSeconds() {
```

Comment:
qq, what is the diff between `expiry` and `expiresInSeconds`?

Reply:
One in the date time of the expiration, one of them is the time in seconds.

Its a little strange to only provide the expiry, which is the value on the token - because when you passed the parameter you were indicating the value in seconds - however, if you are downstream client you'd want to know when you should stop using the token so you'd want the computed DateTime value as well.  Hence the two values.  This is an internal only representation, we can iterate on what the different interfaces are that are exposed at OpenSearch to make this clearer for consumers.